fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBOrQueryToUnionTest.java

```diff
@@ -1077,6 +1092,320 @@ void testOrQueryOrdered() throws Exception {
+                    .asList()
+                    .join();
+
+            // Validate that the queried records contain all expected values
+            assertThat(queried, containsInAnyOrder(expected));
+
+            // Validate sort
+            Integer maxNumValue3 = null;
+            for (TestRecords1Proto.MySimpleRecord queriedRec : queried) {
+                if (maxNumValue3 != null) {
+                    assertThat(maxNumValue3, lessThanOrEqualTo(queriedRec.getNumValue3Indexed()));
+                }
+                maxNumValue3 = queriedRec.getNumValue3Indexed();
+            }
+        }
+
+        // Update the plan configuration to defer fetches. With the old planner, this moves the fetch to before the queries are executed.
+        // With the new planner, this is already the default behavior
+        setDeferFetchAfterUnionAndIntersection(true);
+        // Fetch(Covering(Index(multi_index [[odd, 0],[odd, 0]])) ∪[Field { 'num_value_3_indexed' None}, Field { 'rec_no' None}] Covering(Index(multi_index [[odd, 2],[odd, 2]])))
+        plan = planner.plan(query);
+        if (planner instanceof RecordQueryPlanner) {
+            final BindingMatcher<? extends RecordQueryPlan> planMatcher = fetchFromPartialRecordPlan(
+                    RecordQueryPlanMatchers.unionOnExpressionPlan(
+                            coveringIndexPlan().where(indexPlanOf(indexPlan().where(indexName("multi_index")).and(scanComparisons(range("[[odd, 0],[odd, 0]]"))))),
+                            coveringIndexPlan().where(indexPlanOf(indexPlan().where(indexName("multi_index")).and(scanComparisons(range("[[odd, 2],[odd, 2]]"))))))
+                            .where(comparisonKey(concat(field("num_value_3_indexed"), primaryKey("MySimpleRecord"))))
+            );
+            assertMatchesExactly(plan, planMatcher);
+
+            assertEquals(-1754668377, plan.planHash(CURRENT_LEGACY));
+            assertEquals(1632504435, plan.planHash(CURRENT_FOR_CONTINUATION));
+        } else {
+            final BindingMatcher<? extends RecordQueryPlan> planMatcher =
+                    fetchFromPartialRecordPlan(
+                            RecordQueryPlanMatchers.unionOnValuesPlan(
+                                    coveringIndexPlan()
+                                            .where(indexPlanOf(indexPlan().where(indexName("multi_index")).and(scanComparisons(range("[[odd, 0],[odd, 0]]"))))),
+                                    coveringIndexPlan()
+                                            .where(indexPlanOf(indexPlan().where(indexName("multi_index")).and(scanComparisons(range("[[odd, 2],[odd, 2]]"))))))
+                                    // note that the comparison key is on num_value_3_indexed + the primary key
+                                    .where(comparisonKeyValues(exactly(fieldValueWithFieldNames("num_value_3_indexed"), fieldValueWithFieldNames("num_value_2"), fieldValueWithFieldNames("rec_no")))));
```

Comment:
This is strange as we should drop the `num_value_2` part.

Reply:
Maybe I am misremembering the details of planning this in the Cascades Planner but we should track that. Can you please open an issue for me?

Reply:
Discussed this a bit offline, but we believe the difference here is that the Cascades planner includes the equality bound field here because it can make the result of the union usable in more cases (e.g., if there's a requirement that the plan be ordered by that field that the planner doesn't (yet) know about). In any case, the additional field is harmless (it effectively means that it will consume one leg of the union first before the other leg), but not necessary. There are other cases where one of the equality bound fields is also an ordering key where the inclusion of the field is _required_, and there are already some tests for that